spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/web/client/cookies/JdkPerInstanceCookieStoreTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2014-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.codecentric.boot.admin.server.web.client.cookies;
+
+import java.io.IOException;
+import java.net.CookieHandler;
+import java.net.URI;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import de.codecentric.boot.admin.server.domain.entities.Instance;
+import de.codecentric.boot.admin.server.domain.values.InstanceId;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+class JdkPerInstanceCookieStoreTest {
+
+	private static final Instance INSTANCE = Instance.create(InstanceId.of("i"));
+
+	private CookieHandler cookieHandler;
+
+	private JdkPerInstanceCookieStore store;
+
+	@BeforeEach
+	void setUp() throws Exception {
+		store = spy(new JdkPerInstanceCookieStore());
+		cookieHandler = mock(CookieHandler.class);
```

Comment:
Not sure why you mock the jdk classes here, wouldn't it be easier to just use the real implementation?
So no spy/mock would be needed, you could just call put and get on your implementation and thats it?

But would be ok to leave it that way now, would just be interested in the reason.

Reply:
The CookieManager (impl of CookieHandler) has a lot of rules and conditions inside to match and return cookies. As I only want to test the impl of JdkPerInstanceCookieStore it's only important that the CookieHandler returnes the cookies. Goal is to test the conversion from what the CookieHandler returnes to what the JdkPerInstanceCookieStore returnes. Spying was only necessary because I didn't wanted to to make the creation of CookieManager configurable in production.